core/lib/dal/src/consensus/proto/mod.proto

```diff
@@ -114,3 +116,8 @@ message PaymasterParams {
   optional bytes paymaster_address = 1; // required; H160
   optional bytes paymaster_input = 2; // required
 }
+
+message AttestationStatus {
+  optional roles.validator.GenesisHash genesis = 1; // required
+  optional uint64 next_batch_to_attest = 2; // required
+}
```

Comment:
Is the purpose of these protobuf definitions solely to serialise an `AttestationStatus` record to JSON? Curious, what would be the downside of relying on `#[derive(serde::Serialize)]` instead?

Reply:
a) lack of backward compatibility control over the format
b) GenesisHash doesn't implement Serialize and it shouldn't because then the internal structure becomes public API.
If you don't like it, I can do a wrapper for just GenesisHash instead.

Reply:
Could we not manually implement `Serialize` for `GenesisHash`? Not that it's needed, just asking. 
I thought it might be backward compatibility. It's certainly an interesting world, because Protobuf would be backward compatible due to the numeric tags, and `serde` supports aliases for fields, but our CI rejects any renamings for example. It's nice to have Protobuf when we use binary _and_ JSON, I was just a little surprised at the level of ceremony needed to return a simple untyped JSON structure. No need to do anything, just wanted to understand why we do it.

Reply:
I guess you could but you don't know exactly which format to apply in JSON. In the past I used the [is_human_readable](https://docs.rs/serde/latest/serde/trait.Serializer.html#method.is_human_readable) method to see if I should apply the binary format or the string based one. We could use `TextFmt` when it's supposed to be human readable and `ByteFmt` when it's not, to derive `Serialize` and `Deserialize` when both of these traits are implemented for a type.

Reply:
Argh, of course it doesn't matter if we can't use it because the dependency is inaccessible 🤦

Reply:
we maintain both WIRE and WIRE_JSON compatibility: https://buf.build/docs/breaking/rules . Technically we could limit it to WIRE_JSON for some packages, but that would require totally isolating the stuff which is JSON encodable and binary encodable, which is not worth it as it disallows interoperability. In era-consensus protocol we use binary because it is already there, here we use JSON, well, because it is a json rpc server.